### PR TITLE
Adding test environment documentation to README

### PR DESCRIPTION
**Pull request by tgross**
This PR includes README updates for our test environment configuration. The idea here is to show someone how they'd use this module to build new test containers and how our Shippable environment works. The latter is still WIP.

Requires https://github.com/autopilotpattern/testing/pull/5 to be accurate but is non-overlapping so by having it as a separate PR branch it's easier to review.

cc @misterbisson

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,3 +5,104 @@
...
 See the [API documentation](https://github.com/autopilotpattern/testing/blob/master/API.md) for details on the methods made available on the `AutopilotPatternTest` class.
+
+## Test environments
+
+The minimal environment for an application that uses the `testcases` module is:
+
+- a Docker client.
+- a Docker deployment target such as Triton or Docker for Mac.
+- Python 3
+- installing `requirements.txt` via `pip3` (this will include Docker Compose).
```
Comment (misterbisson) [inline README.md:16]: It's stated elsewhere, but a Dockerfile is required as well.

Comment (misterbisson) [inline README.md:16]: Understandably, you might not be considering that part of the environment. 

There's an implied knowledge that the user understands they have to bring their own tests. Should that be more explicitly stated up top?

Comment (tgross) [inline README.md:16]: Probably.

File: README.md [hunk 2]
```diff
@@ -5,3 +5,104 @@
...
+  hub:
+    - integrationName: DockerHub
+      type: docker
+
+  key:
+    - integrationName: MyTritonTestingKey
+      type: ssh-key
+```
+
+We can't mount our credentials on our laptops to the build container on Shippable, so we need to use the ssh key integration to create a Triton profile using that key. While we could create a Triton profile and commit it to the GitHub repo safely (the private key isn't in our repo), it's better to dynamically generate it when we run the build. This makes it easy for anyone to set up their own Shippable test pipeline without hard-coded users.
```
Comment (misterbisson) [inline README.md]: > the private key isn't in our repo

I think you mean that the key isn't part of the profile that's committed to the repo, but it took me some time to parse.

Comment (tgross) [inline README.md]: Right. Will clean that language up.

**Comment by misterbisson** [comment]
It looks like you're on the right path, merge when ready.

**Pull request merged**